src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -990,6 +1471,36 @@ public static bool TryTakeTransition(SymbolicRegexMatcher<TSet> matcher, ref Cur
                 return false;
             }
 
+            /// <summary>Transition function that only considers DFA state id</summary>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
style wise @stephentoub when there's a single caller and we force inline, do we prefer a nested method? (actually does the JIT even inline those today?)

Reply:
It doesn't need to be nested, but it shouldn't be public if it's only used by this class.